### ISSUE TITLE
PC keyboard only paritally usable?

### ISSUE THREAD

**Issue by bnsmb** [bk138/droidVNC-NG#251]
**Describe the bug**

The PC keyboard can not be used in the VNC Viewer when connected to a droidVNC-NG (except the keyboard shortcuts  like Ctrl-Shift-Esc, etc)

I guess from reading the docs this is "working as designed" but I'm not sure about it.

**Your environment (please complete the following information):**

 - droidVNC-NG version:   2.7.3
 - Android version: OmniROM 15 (= Android 15) and stock Android 14 on a Lenovo Tab Plus
 - Client-side OS and version: Linux, Fedora 40
 - VNC client and version:

tigervnc-1.14.1-2.fc40.x86_64
novnc-1.4.0-3.fc40.noarch

**Comment by bk138** [comment]
You should be able to enter text into text fields, but this needs the Input Accessibility Service to be connected. The UI takes you there on startup. Are all permissions set to green GRANTED displays?

**Comment by bnsmb** [comment]
>> Are all permissions set to green GRANTED displays?

yes (creating a screen shot to prove that seems to be disabled in the app ..)

**Comment by bk138** [comment]
> > > Are all permissions set to green GRANTED displays?
> 
> yes (creating a screen shot to prove that seems to be disabled in the app ..)

Thanks for the quick reply. The only option then is to collect logcat output so we can try to find any anomaly.

**Comment by bnsmb** [comment]
ok, I collect the logcat then later today or tomorrow

**Comment by bnsmb** [comment]
any special parameter for logcat? Or just start logcat;  start VNC Server, connnect via VNC client and push keys on the PC keyboard?

**Comment by bk138** [comment]
> any special parameter for logcat? Or just start logcat; start VNC Server, connnect via VNC client and push keys on the PC keyboard?

Yeah full log, we can parse through the rest.

**Comment by bnsmb** [comment]
done -- I tested it on OmniROM 13, 14 and 15 (the version number is also the Android version). The error exists on all three versions

On all tree Android versions I sometimes lost the connection with this error message:

![image](https://github.com/user-attachments/assets/44e5f50a-2750-43ee-8319-7450ea79184a)


A logcat for each Android version is in the attached zip .  

The last log entries are the messages I got after pressing a key on the PC keyboard, e.g.:

12-11 14:14:51.318  8316  8678 D InputService: onKeyEvent: keysym 100 down 0 by client -5476376628669444000

[droidVNC-ng_logcat.zip](https://github.com/user-attachments/files/18096096/droidVNC-ng_logcat.zip)

**Comment by bk138** [comment]
I have a slight idea. Can you pls try with the following build?
[test.zip](https://github.com/user-attachments/files/18096578/test.zip)

**Comment by bk138** [comment]
@bnsmb are these results with the test app from https://github.com/bk138/droidVNC-NG/issues/251#issuecomment-2536045714? Please note that keyboard entry only works with TextViews, not Canvas.

**Comment by bnsmb** [comment]
> @bnsmb are these results with the test app from https://github.com/bk138/droidVNC-NG/
> issues/251#issuecomment-2536045714?

yes

>> Please note that keyboard entry only works with TextViews, not Canvas.

Hmm, and how can I know if an app uses Canvas?

**Comment by bk138** [comment]
> > > Please note that keyboard entry only works with TextViews, not Canvas.
> 
> Hmm, and how can I know if an app uses Canvas?

I meant EditText, sorry. But the same applies. EditText is basically the standard text entry widget drawn where the framework is responsible for drawing and input; with Canvas OTOH the app is responsible for drawing and getting input. tl;dr_ browser keyboard input: yes: terminal emulator: rather not, unless they're using EditText.

**Comment by bnsmb** [comment]
ah, okay

so the not working keyboard in the terminal app is most probably working as designed .

Anyway, the most important feature of droidVNC-NG for me is that it doesn't need root access and it can start automatically after the phone has booted, so I can then activate adb over WiFi remotely  and connect to adb over WiFi.

**Comment by bk138** [comment]
Okay, I'll close this then with some extended documentation of the keyboard input.

**Comment by bk138** [comment]
Fixed since #208.
